src/Elliptic/Systems/Poisson/BoundaryConditions/Robin.cpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Poisson/BoundaryConditions/Robin.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Poisson::BoundaryConditions::detail {
+
+RobinImpl::RobinImpl(const double dirichlet_weight, const double neumann_weight,
+                     const double constant, const Options::Context& context)
+    : dirichlet_weight_(dirichlet_weight),
+      neumann_weight_(neumann_weight),
+      constant_(constant) {
+  if (dirichlet_weight == 0. and neumann_weight == 0.) {
+    PARSE_ERROR(
+        context,
+        "Either the dirichlet_weight or the neumann_weight must be non-zero "
+        "(or both).");
+  }
+}
+
+double RobinImpl::dirichlet_weight() const noexcept {
+  return dirichlet_weight_;
+}
+double RobinImpl::neumann_weight() const noexcept { return neumann_weight_; }
+double RobinImpl::constant() const noexcept { return constant_; }
+
+void RobinImpl::apply(const gsl::not_null<Scalar<DataVector>*> field,
+                      const gsl::not_null<Scalar<DataVector>*>
+                          n_dot_field_gradient) const noexcept {
+  if (neumann_weight_ == 0.) {
+    get(*field) = constant_ / dirichlet_weight_;
+  } else {
+    get(*n_dot_field_gradient) =
+        (constant_ - dirichlet_weight_ * get(*field)) / neumann_weight_;
+  }
+}
+
+void RobinImpl::apply_linearized(
+    const gsl::not_null<Scalar<DataVector>*> field_correction,
+    const gsl::not_null<Scalar<DataVector>*> n_dot_field_gradient_correction)
+    const noexcept {
+  if (neumann_weight_ == 0.) {
+    get(*field_correction) = 0.;
```

Comment:
why are the constants not part of the linearization?

Reply:
This is the linearization w.r.t. to `u` and `d/du c = 0`